Source/VungleAdapter.swift

```diff
@@ -75,37 +75,50 @@ final class VungleAdapter: PartnerAdapter {
         completion(.success(["bid_token": bidToken]))
     }
     
-    /// Indicates if GDPR applies or not and the user's GDPR consent status.
-    /// - parameter applies: `true` if GDPR applies, `false` if not, `nil` if the publisher has not provided this information.
-    /// - parameter status: One of the `GDPRConsentStatus` values depending on the user's preference.
-    func setGDPR(applies: Bool?, status: GDPRConsentStatus) {
+    /// Indicates that the user consent has changed.
+    /// - parameter consents: The new consents value, including both modified and unmodified consents.
+    /// - parameter modifiedKeys: A set containing all the keys that changed.
+    func setConsents(_ consents: [ConsentKey: ConsentValue], modifiedKeys: Set<ConsentKey>) {
+        // GDPR
         // See https://support.vungle.com/hc/en-us/articles/360048572411
-        if applies == true, status != .unknown {
-            let gdprStatus = status == .granted ? true : false
-            VunglePrivacySettings.setGDPRStatus(gdprStatus)
-            log(.privacyUpdated(setting: "GDPR Status", value: gdprStatus))
+        if modifiedKeys.contains(partnerID) || modifiedKeys.contains(ConsentKeys.gdprConsentGiven) {
+            let consent = consents[partnerID] ?? consents[ConsentKeys.gdprConsentGiven]
```

Comment:
Hmm, it might be nice if there was some helper so this entire block wasn't duplicated in every adapter, e.g.

```
if let consent = gdprConsent(for partner: partnerID) {
  // Do something
}
```

But not a big deal.

Reply:
After looking at more adapters, there's enough variation between adapters that this doesn't really make as much sense.